### PR TITLE
fix: boolean option should work with strict

### PR DESCRIPTION
**Pull request by jly36963**
Addresses: https://github.com/yargs/yargs/issues/1694

When defining an option with `yargs.option`, this line is called:

```js
this.#options.key[key] = true;
```

This affects the behavior in`validation > unknownArguments > isValidAndSomeAliasIsNotNew`. Without the tracking of set keys, `unknownArguments` will fail (as described in the issue).

I turned that line into a reusable function so that `yargs.option` and the array/boolean/count/number/string methods have consistent behavior when using `yargs.strict`

### CODE REVIEW

**Comment by bcoe** [comment]
@jly36963 this looks like a great solution 👏 It's the end of the day and my brain is a little tired, so I'm going to wait to review fully.

In the meantime, I'm going to [release this](https://github.com/yargs/yargs/pull/1984), which has several of your patches in it. 

Thanks for the contributions.

**Comment by jly36963** [comment]
> Thanks for the contributions.

For sure! Glad I can help 👍

**Comment by bcoe** [comment]
@jly36963 bother you to rebase?

File: lib/yargs-factory.ts [hunk 1]
```diff
@@ -2206,6 +2214,15 @@ export class YargsInstance {
   [kSetHasOutput]() {
     this.#hasOutput = true;
   }
+  [kTrackManuallySetKeys](keys: string | string[]) {
+    if (typeof keys === 'string') {
+      this.#options.key[keys] = true;
```
Comment (bcoe) [inline lib/yargs-factory.ts:2219]: doesn't the `this[kPopulateParserHintArray]('boolean', keys);` function also populate `this.#options`?

I'm wondering why we need this additional `kTrackManuallySetKeys`? I'm probably missing something.

**Review by bcoe — CHANGES_REQUESTED** [review]

File: lib/yargs-factory.ts (hunk 1 above)
Comment (jly36963) [reply lib/yargs-factory.ts:2219]: ## Example code

```js
yargs
  .command({
    command: "cmd",
    desc: "cmd desc",
    builder: (yargs) =>
      yargs
        .option("opt1", {
          desc: "opt1 desc",
          type: "boolean",
        })
        .option("opt2", {
          desc: "opt2 desc",
          boolean: true,
        })
        .array("opt3")
        .boolean("opt4")
        .count("opt5")
        .number("opt6")
        .string("opt7")
        .fail(() => {
          console.log('failure!!!')
        }),
    handler: (argv) => {
      console.log(argv);
    },
  })
  .help()
  .strict()
  .parse("cmd --opt1 --opt2 --opt3 foo bar baz --opt4 --opt5 --opt6 3 --opt7 oof");
  ```
## Reason for the change

`yargs.option` has this line:

```js
this.#options.key[keys] = true;
```
This line populates `options.key`. 

Before my fix, calling `yargs.boolean` (or array, count, string, number, etc) was missing this behavior.

To highlight the difference in behavior between `.option()` and `.boolean()`, I added opt1 and opt2 using `.option()`. In the "Without fix" section below, you should see that `options.key` has opt1 and opt2, but is missing the others.

## Explanation 

I logged options (`yargs.getOptions()`) inside of `validation.isValidAndSomeAliasIsNotNew` 

Without the fix, `options.key` is missing keys, so `isValidAndSomeAliasIsNotNew` fails.



### Without fix:

```js
{
  local: [],
  configObjects: [],
  array: [ 'opt3' ],
  boolean: [ 'version', 'help', 'opt1', 'opt2', 'opt4' ],
  string: [ 'opt7' ],
  skipValidation: [],
  count: [ 'opt5' ],
  normalize: [],
  number: [ 'opt6' ],
  hiddenOptions: [],
  narg: {},
  key: { version: true, help: true, opt1: true, opt2: true },
  alias: {},
  default: {},
  defaultDescription: {},
  config: {},
  choices: {},
  demandedOptions: {},
  demandedCommands: {},
  deprecatedOptions: {},
  envPrefix: undefined,
  __: [Function: bound __],
  configuration: {}
}
```

### WIth fix

```js
{
  local: [],
  configObjects: [],
  array: [ 'opt3' ],
  boolean: [ 'version', 'help', 'opt1', 'opt2', 'opt4' ],
  string: [ 'opt7' ],
  skipValidation: [],
  count: [ 'opt5' ],
  normalize: [],
  number: [ 'opt6' ],
  hiddenOptions: [],
  narg: {},
  key: {
    version: true,
    help: true,
    opt1: true,
    opt2: true,
    opt3: true,
    opt4: true,
    opt5: true,
    opt6: true,
    opt7: true
  },
  alias: {},
  default: {},
  defaultDescription: {},
  config: {},
  choices: {},
  demandedOptions: {},
  demandedCommands: {},
  deprecatedOptions: {},
  envPrefix: undefined,
  __: [Function: bound __],
  configuration: {}
}
```

**Review by bcoe — APPROVED** [review]

**Pull request merged**